java/arcs/tools/PlanGenerator.kt

```diff
@@ -0,0 +1,168 @@
+package arcs.tools
+
+import arcs.core.data.CountType
+import arcs.core.data.CreatableStorageKey
+import arcs.core.data.EntityType
+import arcs.core.data.FieldName
+import arcs.core.data.FieldType
+import arcs.core.data.Plan
+import arcs.core.data.Recipe
+import arcs.core.data.Schema
+import arcs.core.data.SchemaFields
+import arcs.core.data.SchemaName
+import arcs.core.data.TupleType
+import arcs.core.data.TypeVariable
+import arcs.core.storage.StorageKeyParser
+import arcs.core.type.Type
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.FileSpec
+import com.squareup.kotlinpoet.PropertySpec
+import com.squareup.kotlinpoet.buildCodeBlock
+
+fun Recipe.toGeneration(builder: FileSpec.Builder) {
+    val handles = this.handles.values.map { it.toGeneration(name.orEmpty()) }
+    val ctx = mapOf(
+        "plan" to Plan::class,
+        "handles" to handles.toGeneration("%N"),
+        // TODO(161940699) Generate particles
+        "particles" to listOf<Recipe.Particle>().toGeneration(),
+        // TODO(161940729) Generate Annotations
+        "annotations" to listOf<Annotation>().toGeneration()
+    )
+    val plan = PropertySpec.builder("${name}Plan", Plan::class)
+        .initializer(buildCodeBlock {
+            addNamed("""
+                %plan:T(
+                    particles = %particles:L,
+                    handles = %handles:L,
+                    annotations = %annotations:L
+                )
+                """.trimIndent(), ctx)
```

Comment:
Won't this trim all of the indentation? Is that WAI? You could switch this to trimMargin and keep the indentation for the particles, handles, and annotations line?